panel_grid.py

```diff
@@ -75,28 +79,27 @@ def show_table(self, week, tab):
         """
         Creates a Table/Grid showing days, activities, checkboxes, ...
         """
-        if len(week) < 7:
-            current_week = [i.date.strftime("%A") for i in week]
-            hybrid_week = self.t.get_remaining(len(week)) + current_week
-        else:
-            hybrid_week = []
-
-        for i in range(len(self.activities)+1):              # for rows in activities
-            for j in range(len(self.days)+1):                # for cols in days
+        days = self.t.get_days_names(week)
+        activities = self.t.get_activities_names(week)
+        
+        for i in range(len(activities)+1):              # for rows in activities
+            for j in range(len(days)+1):                # for cols in days
                 if i == 0 and j == 0:                                  # if its first cell, add empty cell
                     self.labeling(tab, i, j, Label(tab, text=" "))
                 elif i == 0:                                           # adding the name of the day
-                    self.labeling(tab, i, j, Label(tab, text=self.days[j-1]))
+                    print( "This is j: " + str(j) + ", this is i: " + str(i))
+                    self.labeling(tab, i, j, Label(tab, text=days[j-1]))
+                elif activities[0] == " ":                             # if there are no activities 
+                    self.labeling(tab, i, j, Label(tab, text=" "))
                 elif j == 0:                                           # adding the name of the activity
-                    self.labeling(tab, i, j, Label(tab, text=self.activities[i-1]))
+                    self.labeling(tab, i, j, Label(tab, text=activities[i-1]))
                 else:                                                  # adding the checkboxes
-                    if hybrid_week:
-                        if hybrid_week[j-1] in current_week:
-                            self.labeling(tab, i, j, Checkbutton(tab))
-                        else:
-                            self.labeling(tab, i, j, Checkbutton(tab, state=DISABLED))
```

Comment:
Cleaned up that horror of _hybrid-week_ , not absolutely necessary to have all days of the week lined up...
3rd problemo: 
Your mission, should you choose to accept it: **adding mood to show_table + making sure there is a get_moods in timeline + making sure mood is stored like activities **